src/main/java/org/folio/rest/impl/InvoiceLineHelper.java

```diff
@@ -95,71 +77,48 @@ public class InvoiceLineHelper extends AbstractHelper {
   @Autowired
   private BudgetExpenseClassService budgetExpenseClassService;
 
-  public InvoiceLineHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
-    this(getHttpClient(okapiHeaders), okapiHeaders, ctx, lang);
-    SpringContextUtil.autowireDependencies(this, Vertx.currentContext());
-  }
-
-  public InvoiceLineHelper(HttpClientInterface httpClient, Map<String, String> okapiHeaders, Context ctx, String lang) {
-    super(httpClient, okapiHeaders, ctx, lang);
-    this.protectionHelper = new ProtectionHelper(httpClient, okapiHeaders, ctx, lang);
+  public InvoiceLineHelper(Map<String, String> okapiHeaders, Context ctx) {
+    super(okapiHeaders, ctx);
+    this.protectionHelper = new ProtectionHelper(okapiHeaders, ctx);
     this.adjustmentsService = new AdjustmentsService();
     this.validator = new InvoiceLineValidator();
-    this.restClient = new RestClient();
+    SpringContextUtil.autowireDependencies(this, ctx);
   }
 
-  public CompletableFuture<InvoiceLineCollection> getInvoiceLines(int limit, int offset, String query) {
+  public Future<InvoiceLineCollection> getInvoiceLines(int limit, int offset, String query) {
     return protectionHelper.buildAcqUnitsCqlExprToSearchRecords(INVOICE_LINES)
-      .thenCompose(acqUnitsCqlExpr -> {
+      .compose(acqUnitsCqlExpr -> {
         String queryParam;
         if (isEmpty(query)) {
-          queryParam = getEndpointWithQuery(acqUnitsCqlExpr, logger);
+          queryParam = getEndpointWithQuery(acqUnitsCqlExpr);
         } else {
-          queryParam = getEndpointWithQuery(combineCqlExpressions("and", acqUnitsCqlExpr, query), logger);
+          queryParam = getEndpointWithQuery(combineCqlExpressions("and", acqUnitsCqlExpr, query));
         }
-        String endpoint = String.format(GET_INVOICE_LINES_BY_QUERY, limit, offset, queryParam, lang);
-        return getInvoiceLineCollection(endpoint);
+        String endpoint = String.format(GET_INVOICE_LINES_BY_QUERY, limit, offset, queryParam);
+        return invoiceLineService.getInvoiceLines(endpoint, buildRequestContext());
       });
   }
 
-  public CompletableFuture<List<InvoiceLine>> getInvoiceLinesByInvoiceId(String invoiceId) {
-    String query = getEndpointWithQuery(String.format(QUERY_BY_INVOICE_ID, invoiceId), logger);
+  public Future<List<InvoiceLine>> getInvoiceLinesByInvoiceId(String invoiceId) {
```

Comment:
Can we rewrite this method only with this 1 line?
    return invoiceLineService.getInvoiceLinesByInvoiceId(invoiceId, buildRequestContext()).map(InvoiceLineCollection::getInvoiceLines);

Reply:
done